assets/js/translator.js

```diff
@@ -1242,6 +1265,36 @@ function autoSelectDstLang() {
     $('#dstLangSelect').val(curDstLang).change();
 }
 
+// Check if the Language pair Exist or Not
+function checkLangPairAvailable(lang) {
+    if(languages[lang] !== undefined) {
+        var checkDstLang = pairs[iso639CodesInverse[lang]][0];
+        return checkDstLang !== undefined;
+    }
+    else {
+        return false;
+    }
+}
+
+
+// Set curSrcLang as locale or first available pair, if locale not available
+function noPrefSrcLang() {
+    if(checkLangPairAvailable(iso639Codes[locale])) {
+        curSrcLang = locale;
+    }
+    else {
+        for(var srcLang in pairs) {
+            curSrcLang = srcLang;
+            break;
+        }
+    }
+}
+
+// Get the language code if - exist
+function getLangCode(lang) {
+    return lang.split('-', 2)[0];
```

Comment:
Apertium actually supports language variants so we shouldn't delete the second part. Instead we should convert e.g. `en-US` to `en_US`. The latter is what Apertium uses.

Reply:
done. :)